src/components/TraceDetail.tsx

```diff
@@ -149,7 +151,7 @@ async function loadMoreSpans(
   const end = mkUnixEpochFromNanoSeconds(span.endTimeUnixNano) + 1;
   // See https://github.com/grafana/tempo/issues/5435
   const data = await search(datasourceUid, q, start, end, 4294967295);
-  return await extractSpans(idToLevelMap, traceId, datasourceUid, data);
+  return (await extractSpans(idToLevelMap, traceId, datasourceUid, data)).spans;
```

Comment:
This is a smell really. You are only using `spans` of `extractSpans`, which hints that your function does more than one thing.